lit_parrot/model.py

```diff
@@ -227,14 +264,20 @@ def forward(
                 # shift 1 position to the left
                 cache_k = torch.roll(cache_k, -1, dims=2)
                 cache_v = torch.roll(cache_v, -1, dims=2)
+            padding_idx = cache_k.size(2) - 1  # send padding data to a padding index, doesn't matter which
+            padding = T - input_pos.size(0)
+            input_pos = torch.cat(
+                (input_pos, torch.full((padding,), padding_idx, device=input_pos.device, dtype=input_pos.dtype))
+            )
             k = cache_k.index_copy(2, input_pos, k)
             v = cache_v.index_copy(2, input_pos, v)
             kv_cache = k, v
 
-        # efficient attention using Flash Attention CUDA kernels
-        y = F.scaled_dot_product_attention(
```

Comment:
I was getting nans using `scaled_dot_product_attention` with padding. I will try to repro it and report it to PyTorch